semgrep/semgrep/metric_manager.py

```diff
@@ -46,8 +46,23 @@ def __init__(self) -> None:
 
         self._send_metrics = False
 
-    def set_project_hash(self, project_hash: Optional[str]) -> None:
-        self._project_hash = project_hash
+    def set_project_hash(self, project_url: Optional[str]) -> None:
+        """
+        Standardizes url then hashes
+        """
+        if project_url is None:
+            self._project_hash = None
+        else:
+            parsed_url = urlparse(project_url)
```

Comment:
Should we catch a `ValueError` here, and perform some error handling (return `None`?), or would we rather fail loudly? The [`urlparse`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse) docs state when this exception might occur.

Reply:
maybe catch and re-raise a more helpful error? but i'm also fine with skipping this for now for the sake of expediency

Reply:
Caught and falls back to hashing the entire URL